kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/BankParametersAccessServiceImpl.java

```diff
@@ -0,0 +1,759 @@
+            allowedBaiTypes.addAll(getSpecialBaiTypes());
+        }
+        return allowedBaiTypes;
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isCheckReconBai(int)
+     */
+    public boolean isCheckReconBai(int bai) {
+        return getCheckReconBaiTypes().contains(bai);
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isExcludedBai(int)
+     */
+    public boolean isExcludedBai(int bai) {
+        return getExcludedBaiTypes().contains(bai);
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isExcludedCustomerRefNbr(String)
+     */
+    public boolean isExcludedCustomerRefNbr(String custRefNumb) {
+        return getExcludedCustomerRefNbr().contains(custRefNumb);
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isExcludedByBaiAndAccount(int, String)
+     */
+    public boolean isExcludedByBaiAndAccount(int bai, String account) {
+        return getExcludedByBaiAndAccount().contains(bai + ":" + account);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isSpecialBai(int)
+     */
+    public boolean isSpecialBai(int bai) {
+        return getSpecialBaiTypes().contains(bai);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isAllowedBai(int)
+     */
+    public boolean isAllowedBai(int bai) {
+        return getAllowedBaiTypes().contains(bai);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getChartBankObjectCodeForDescriptionSearch(String)
+     */
+    public ChartBankObjectCode getChartBankObjectCodeForDescriptionSearch(String description) {
+
+        for (String descKey : getDescriptionSearchChartBankObjectCodeMap().keySet()) {
+            if (description.contains(descKey)) {
+                return descriptionSearchChartBankObjectCodeMap.get(descKey);
+            }
+        }
+        //not found, returning null;
+        return null;
+    }
+
+    /*
+     * Internal method that populates a Map with description as key and ChartBankObjectCode as values from the ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION parameter
+     */
+    protected Map<String, ChartBankObjectCode> getDescriptionSearchChartBankObjectCodeMap() {
+        if (descriptionSearchChartBankObjectCodeMap == null) {
+            descriptionSearchChartBankObjectCodeMap = new HashMap<String, ChartBankObjectCode>();
+            String acountObjByDescrParm = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION);
+            String[] acountObjByDescrList = acountObjByDescrParm.split(KFSConstants.BankTransactionsParameters.BANK_PARAMETER_DELIM);
+            if (acountObjByDescrList == null || acountObjByDescrList.length == 0) {
```

Comment:
A nit, but acountObjByDescrList cannot be null here, otherwise line 485 would blow up.